internal/upstreamldap/upstreamldap.go

```diff
@@ -455,7 +464,7 @@ func (p *Provider) searchGroupsForUserDN(conn Conn, userDN string) ([]string, er
 		groupAttributeName = distinguishedNameAttributeName
 	}
 
-	var groups []string
+	groups := []string{}
```

Comment:
I think this change means that the function will never return nil anymore, right? Did you already look at all the callers (and their callers, etc.) to make sure that nothing had any assumptions around this function returning nil?

Reply:
This comment made me realize we don't have any integration tests for when we search for groups but know we won't get any back. I added one in [1d384b8](https://github.com/vmware-tanzu/pinniped/pull/975/commits/1d384b894752773ae6cead4573621ace47f71552) and it passes. I did look through and didn't see anything that looked like it would cause trouble but thank you for pointing this out.